crates/typst-library/src/layout/ratio.rs

```diff
@@ -8,7 +8,47 @@ use crate::foundations::{repr, ty, Repr};
 
 /// A ratio of a whole.
 ///
-/// Written as a number, followed by a percent sign.
+/// Written as a number, followed by a percent sign. A common use case is
+/// setting the width or height of a container (e.g., [block], [rect], etc.),
+/// thus it is used as a part of a [relative length](relative) (because
+/// internally containers use relative length for width and height fields):
```

Comment:
I feel like something like this might be better:

```suggestion
/// setting the width or height of a container (e.g., [block], [rect], etc.),
/// as it can be used as part of a [relative length]($relative) to represent
/// a certain percentage of the surrounding container's - or the current
/// page's, if there isn't one - corresponding dimension. For example:
```

Reply:
Nnoo, it sounds very complex and weird. Why there is "-" before "or"? "container's" what? I completely don't understand the "if there isn't one - corresponding dimension" part. Corresponding to what? There is no way there is nothing outside: there is always page dimensions, if there is no other container wrappers.

Reply:
I changed it a little in the new commit.

Reply:
If this is resolved, mark it as so.